components/camera/replaypcd/replaypcd_utils_test.go

```diff
@@ -0,0 +1,193 @@
+// stored in a mockCloudConnectionService.
+func createMockCloudDependencies(ctx context.Context, t *testing.T, logger golog.Logger, b bool) (resource.Dependencies, func() error) {
+	listener, err := net.Listen("tcp", "localhost:0")
+	test.That(t, err, test.ShouldBeNil)
+	rpcServer, err := rpc.NewServer(logger, rpc.WithUnauthenticated())
+	test.That(t, err, test.ShouldBeNil)
+
+	test.That(t, rpcServer.RegisterServiceServer(
+		ctx,
+		&datapb.DataService_ServiceDesc,
+		&mockDataServiceServer{},
+		datapb.RegisterDataServiceHandlerFromEndpoint,
+	), test.ShouldBeNil)
+
+	go rpcServer.Serve(listener)
+
+	conn, err := viamgrpc.Dial(ctx, listener.Addr().String(), logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	mockCloudConnectionService := &cloudinject.CloudConnectionService{
+		Named: cloud.InternalServiceName.AsNamed(),
+		Conn:  conn,
+	}
+	if !b {
+		mockCloudConnectionService.AcquireConnectionErr = errors.New("cloud connection error")
+	}
+
+	r := &inject.Robot{}
+	rs := map[resource.Name]resource.Resource{}
+	rs[cloud.InternalServiceName] = mockCloudConnectionService
+	r.MockResourcesFromMap(rs)
+
+	return resourcesFromDeps(t, r, []string{cloud.InternalServiceName.String()}), rpcServer.Stop
+}
+
+// createNewReplayPCDCamera will create a new replay_pcd camera based on the provided config with either
+// a valid or invalid data client.
+func createNewReplayPCDCamera(ctx context.Context, t *testing.T, replayCamCfg *Config, validDeps bool,
+) (camera.Camera, func() error, error) {
+	logger := golog.NewTestLogger(t)
+
+	// var resources resource.Dependencies
+	// var closeRPCFunc func() error
+	// if validDeps {
+	resources, closeRPCFunc := createMockCloudDependencies(ctx, t, logger, validDeps)
+	// }
+
+	cfg := resource.Config{ConvertedAttributes: replayCamCfg}
+	cam, err := newPCDCamera(ctx, resources, cfg, logger)
+
+	return cam, closeRPCFunc, err
+}
+
+// resourcesFromDeps returns a list of dependencies from the provided robot.
+func resourcesFromDeps(t *testing.T, r robot.Robot, deps []string) resource.Dependencies {
+	t.Helper()
+	resources := resource.Dependencies{}
+	for _, dep := range deps {
+		resName, err := resource.NewFromString(dep)
+		test.That(t, err, test.ShouldBeNil)
+		res, err := r.ResourceByName(resName)
+		if err == nil {
+			// some resources are weakly linked
+			resources[resName] = res
+		}
+	}
+	return resources
+}
+
+// getNextDataAfterFilter returns the next point cloud data to be return based on the provided filter and
+// last returned item.
+func getNextDataAfterFilter(filter *datapb.Filter, last string) (int, error) {
+	// Apply the time-based filter based on the second's value in the start and end fields. Because artifacts
+	// do not have timestamps associated with them but are numerically ordered we can approximate the filtering
+	// by sorting for the files which occur after the start second count and before the end second count.
```

Comment:
I didn't really understand how this worked until I looked at the code. I think a quick example in this comment would make this clearer. Something like:

```
// For example, if there are 15 files in the artifact directory, the start time is 2000-01-01T12:00:10Z 
// and the end time is 2000-01-01T12:00:14Z, we will return files 10-14.
```